site/src/github.rs

```diff
@@ -640,6 +680,56 @@ async fn summarize_run(ctxt: &SiteCtxt, commit: QueuedCommit, is_master_commit:
     message
 }
 
+// TODO: determine how should this work
+// Should this be separate from `deserves_attention`?
```

Comment:
I agree that it could be confusing, but with the way the code is currently structured, I'm not sure if this structure would help much. In order to correctly propagate the `Stat` lineage, we would have to also include it in `ArtifactComparison` which creates the summaries from a given `stat`, otherwise we would need to pass the stat manually, which doesn't guard against mistakes. And furthermore, when we analyze the summaries, we treat instructions/max rss differently, it's not unified at all. The only safety precaution that this would provide is this:
```rust
pub(crate) fn deserves_attention_icount(
    summary: Summaries,
) -> bool {
    assert_eq!(summary.stat, Stat::Instructions);
}

pub(crate) fn deserves_attention_maxrss(
    summary: Summaries,
) -> bool {
    assert_eq!(summary.stat, Stat::MaxRSS);
}
```
which doesn't seem to be worth it.

To make it really bullet-proof, we would need to design e.g. some generics parameter like this:
```rust
struct Summaries<Stat> { primary, secondary, phantom: Stat }

fn deserves_attention_icount(summary: Summaries<Stat::Instructions>) { .. .}
```
but it seems like overkill here.

Making `Stat` an enum is a good idea, I'll do it.